cdap-security-spi/src/main/java/io/cdap/cdap/security/spi/authorization/AuditLogRequest.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright © 2024 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.security.spi.authorization;
+
+import java.util.Queue;
+import javax.annotation.Nullable;
+
+/**
+ * The object that will be sent to Auth Extension which would contain the collections of audit logs
+ * {@link AuditLogContext} related to a Single Call or Operation.
+ * This also contains various other metadata required for Audit Logging.
+ */
+public class AuditLogRequest {
```

Comment:
Please also add `userIp` in `AuditLogRequest`. Should be feasible to get from `SecurityRequestContext`.